homeassistant/components/vesync/light.py

```diff
@@ -175,3 +180,165 @@ def color_temp(self) -> int:
         )
         # ensure value between minimum and maximum Mireds
         return max(self.min_mireds, min(color_temp_value, self.max_mireds))
+
+
+class VeSyncMulticolorLightHA(VeSyncTunableWhiteLightHA, LightEntity):
+    """Representation of a VeSync Multicolor Light device."""
+
+    _attr_supported_color_modes = {ColorMode.HS, ColorMode.COLOR_TEMP}
+
+    @property
+    def color_mode(self) -> ColorMode | str | None:
+        """Return the current color mode."""
+        # get value from pyvesync library api,
+        color_mode_value = self.device.color_mode
+        if color_mode_value.lower() == "white":
+            return ColorMode.COLOR_TEMP
+        if color_mode_value.lower() in ["hsv", "color"]:
+            return ColorMode.HS
+        return ColorMode.ONOFF
```

Comment:
I think (don't really remember, I'll double check) but I think I did that as a forethought about leaving the class more generic and reusable for the future. For example, if that class were assigned to a RGB multicolor bulb (which there were none back then). So the ONOFF mode would work as a fallback. so at least you could turn the bulb ON and OFF.

Reply:
If it's not needed now but sometime in the future, this is a case of YAGNI and should be changed.
If you think it may already be needed it's OK to leave it in.